types/src/vote.rs

```diff
@@ -225,16 +241,29 @@ where
         mut self,
         val: (
             Commitment<LEAF>,
-            (EncodedPublicKey, (EncodedSignature, VoteData<LEAF>, TOKEN)),
+            (EncodedPublicKey, (EncodedSignature, Vec<StakeTableEntry<QCVerKey>>, usize, VoteData<LEAF>, TOKEN)),
         ),
-    ) -> Either<Self, YesNoSignature<LEAF, TOKEN>> {
-        let (commitment, (key, (sig, vote_data, token))) = val;
+    ) -> Either<Self, AssembledSignature> {
+        let (commitment, (key, (sig, entries, node_id, vote_data, token))) = val;
+
+        // Desereialize the sig so that it can be assembeld into a QC
+        let origianl_sig: <BLSOverBN254CurveSignatureScheme as SignatureScheme>::Signature 
+        = bincode_opts().deserialize(&sig.clone().0).unwrap();
```

Comment:
Is there a chance this is deserialized incorrectly due to malformed input? Suppose I send over a bunch of random bits for a signature that serde can't decipher. Could this panic?